### ISSUE TITLE
License clarification

### ISSUE THREAD

**Issue by git-bruh** [Sangwan5688/BlackHole#208]
Hi, some of the license terms mentioned in the [readme](https://github.com/Sangwan5688/BlackHole#license) don't match up with the GPL.

>  1. You can't upload the app on PlayStore/AppStore or any other closed source app repository.
> 2. You can't sell any copied/modified version of the app
> 3. You can't put ads to this app

The GPL imposes no restrictions as the quoted ones, given that you make the modified source code available under the same license and in-fact encourages people to [sell free software](https://www.gnu.org/philosophy/selling.en.html).

I am also not aware of any free software license that restricts reselling, so you would have to create a custom license that has these restrictions. These restrictions  make the app technically non-free software, making it unfit for inclusion into [F-Droid](https://f-droid.org/en/docs/Inclusion_Policy/) since it requires that all apps be distributed under a free software license.

Thanks

**Comment by Sangwan5688** [comment]
The App is under `GPL v3.0` and has the same restrictions as `GPL`. I just added these bullet points 3-4 days ago coz a lot of people are unfamiliar with whole license policy and just create a copy which `isn't` under GPL.
For the first point i.e. uploading on PlayStore/AppStore. If an app is under GPL then any copy/fork of it should be under GPL. I have seen a lot of users who just clone repo compile it and upload it on playstore without any open source copy of the same, which clearly violates GPL. Again for selling I meant the same thing i.e. redistribution of the app under a `non-free` license.
Those bullet points were just made for normal users who aren't familiar with all that stuff. As for users who are familiar with GPL or any other open-source license already know everything and needn't pay attention to it.

**Comment by git-bruh** [comment]
Thanks. I suggest rewording the notice to something like this to avoid such confusion in the future:

```
The following activities MUST NOT be performed unless the app's modified source code is made available on an open source hosting platform like GitHub with proper credits under the license `GPL v3.0 or later`:
1. Uploading the app on PlayStore/AppStore or any other app repository
2. Selling copied/modified versions of the app
3. Adding ads to the app
```

**Comment by Sangwan5688** [comment]
Hmmm, See the overall GPL license is too much to write in readme. And I don't know how to write it in simple words so that no one misunderstands.

```
Being Open Source doesn't mean you can just make a copy of the app and upload it on playstore
or sell a closed source copy of the same.

Read the following carefully:
1. Any copy of a software under GPL must be under same license. So you can't upload the app on a closed source
  app repository like PlayStore/AppStore without distributing the source code.
2. You can't sell any copied/modified version of the app under any "non-free" license.
   You must provide the copy with the original software or with instructions on how to obtain original software,
   should clearly state all changes, should clearly disclose full source code, should include same license
   and all copyrights should be retained.

In simple words, You can ONLY use the source code of this app for `Open Source` Project under `GPL v3.0` or later
with all your source code clearly disclosed on any open source platform like GitHub, with clear instruction on
how to obtain the original software, should clearly state all the changes made and should retain all copyrights.
Use of this software under any "non-free" license is NOT permitted.
```

I hope this looks fine now. I tried to make it less confusing. If there's still some confusion in this lemme know the same

**Comment by pravi** [comment]
You are allowed to upload forks/copies on play store under GPL, play store may have their own rules for clones. As an example see conversations app, there is already two forks/modified copies, blabber.im and moncles chat.

They only need to keep the same license.

**Comment by pravi** [comment]
See github makes a nice summary of GPL 3.0

**Comment by pravi** [comment]
See https://github.com/iNPUTmice/Conversations/blob/master/LICENSE for a short and accurate summary.
